plugins/kotlin-as-java/src/main/kotlin/converters/KotlinToJavaConverter.kt

```diff
@@ -184,8 +185,36 @@ internal fun DFunction.asJava(containingClassName: String): DFunction {
         else sourceSets.map { it to modifier.values.first() }.toMap(),
         parameters = listOfNotNull(receiver?.asJava()) + parameters.map { it.asJava() },
         visibility = visibility.map { (sourceSet, visibility) -> Pair(sourceSet, visibility.asJava()) }.toMap(),
-        receiver = null
-    ) // TODO static if toplevel
+        receiver = null,
+        extra = if (isTopLevel) {
+            extra + extra.mergeAdditionalModifiers(
+                sourceSets.map {
+                    it to setOf(ExtraModifiers.JavaOnlyModifiers.Static)
+                }.toMap()
+            )
+        } else {
+            extra
+        }
+    )
+}
+
+
+internal fun DFunction.asJava(containingClassName: String, isTopLevel: Boolean = false): List<DFunction> {
+    val newName = when {
+        isConstructor -> containingClassName
+        else -> name
+    }
+    val baseFunction = asJava(containingClassName, newName, parameters, isTopLevel)
+    return if (jvmOverloads() == null) {
+        listOf(baseFunction)
+    } else {
+        val functions = mutableListOf(baseFunction)
+        val parameters = this.parameters.toMutableList()
+        while (parameters.removeLastIf { it.extra[DefaultValue] != null }) {
+            functions.add(asJava(containingClassName, newName, parameters, isTopLevel))
+        }
+        functions
```

Comment:
IMHO this could be written like this and then we can remove the `removeLastIf` entirely
```suggestion
        val endParamsWithDefaults = parameters.takeLastWhile { it.extra[DefaultValue] != null }
        val startParamsWithoutDefaults = parameters - endParamsWithDefaults

        endParamsWithDefaults.runningFold(startParamsWithoutDefaults) { acc, param -> acc + param }.map { params -> asJava(containingClassName, newName, params, isTopLevel) }

```

Reply:
I took your advice on this and extracted the logic to another method but changed a little from what you gave, both because your code didn't take the case where there was a non-default param between default params and to ensure the order of generated functions respects what's in the [`JvmOverloads` documentation](https://kotlinlang.org/api/latest/jvm/stdlib/kotlin.jvm/-jvm-overloads/-init-.html)

Reply:
Nice catch! Thanks ☺️